mgnify_pipelines_toolkit/analysis/shared/study_summary_generator.py

```diff
@@ -0,0 +1,263 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2024 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from collections import defaultdict
+import glob
+import logging
+from pathlib import Path
+
+import pandas as pd
+
+from mgnify_pipelines_toolkit.constants.db_labels import TAXDB_LABELS, ASV_TAXDB_LABELS
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i",
+        "--indir",
+        required=True,
+        type=str,
+        help="Input directory to where all the individual analyses subdirectories are if summarising, or where all the summaries are if merging",
+    )
+    parser.add_argument(
+        "-r",
+        "--runs",
+        required=False,
+        type=str,
+        help="File containing successful analyses generated by the pipeline",
+    )
+    parser.add_argument(
+        "-m",
+        "--mode",
+        required=True,
+        type=str,
+        choices=["summarise", "merge"],
+        help="Mode to either summarise analyses (summarise), or merge multiple existing summaries (merge)",
+    )
+    parser.add_argument(
+        "-p", "--prefix", required=True, type=str, help="Prefix to output"
+    )
+
+    args = parser.parse_args()
+
+    indir = args.indir
+    runs = args.runs
+    mode = args.mode
+    prefix = args.prefix
+
+    return indir, runs, mode, prefix
+
+
+def get_tax_file(run_acc, analyses_dir, db_label):
```

Comment:
```suggestion
def get_tax_file(run_acc: str, analyses_dir: Path, db_label: str) -> Union[Path, List[Path]]:
"""
:returns: ....
"""
```
This is a good place for some type hints and a docstring description of the return type, since the return type is conditional.